### DISCUSSION TITLE
Best way to execute this script

### DISCUSSION THREAD

**Discussion by cesarhijarsm** [ktaranov/sqlserver-kit#216] [Q&A] [answered]
Hi, 
I'm a newbie on sql server. What would be the best way to execute this script in order all results save to a file?
I mean, in Oracle we can put ---> spool filename.lst, then execute the statements, and then: spool off and all the results are on that file.

Have SQL Server has similar way?

thank you

**Comment by ktaranov** [comment, marked as answer]
Yes, it is. You can use default sqlcmd utility https://stackoverflow.com/a/425463/2298061

But but I highly recommend mastering a great powershell module https://docs.dbatools.io/Export-DbaXECsv.html

**Reply by cesarhijarsm** [reply]
Thank you for your response, it will be great when I execute one statement. 
Let me explain. I want to execute all this sql script once and the result of all statements save on a txt file. 
How can I do that?
thanks in advanced

**Reply by ktaranov** [reply]
> Thank you for your response, it will be great when I execute one statement. Let me explain. I want to execute all this sql script once and the result of all statements save on a txt file. How can I do that? thanks in advanced

could you provide your script and desire result?

**Comment by cesarhijarsm** [comment]
Hi, 
the script is 
-- SQL Server 2008 R2 Diagnostic Information Queries
-- Glenn Berry 
[SQLServer2008HealthCheck_v01.txt](https://github.com/ktaranov/sqlserver-kit/files/10830060/SQLServer2008HealthCheck_v01.txt)

I will execute this on MSSQL server 2008 
regards,

**Reply by ktaranov** [reply]
```ps
Invoke-DbaDiagnosticQuery -SqlInstance sql2016 -UseSelectionHelper | Export-DbaDiagnosticQuery -Path C:\temp\gboutput
```

I strongly recommend spending 2 hours of your time to master the magnificent dbatools package - it will be the best investment in your developer career!
